test/test_interfaces.py

```diff
@@ -1,4 +1,84 @@
+    # (so it doesn't have to take the 'enabled' argument)
+    # and since it's specific to the launch method, it should probably be
+    # renamed to send_launch_error_analytics.
+    # these tests test its current behavior
+    @mock.patch("requests.post")
+    def test_error_analytics_doesnt_crash_on_connection_error(self, mock_post):
+        mock_post.side_effect = requests.ConnectionError()
+        send_error_analytics(True)
+        mock_post.assert_called()
+        
+    @mock.patch("requests.post")
+    def test_error_analytics_doesnt_post_if_not_enabled(self, mock_post):
+        send_error_analytics(False)
+        mock_post.assert_not_called()
+     
+    @mock.patch("requests.post")   
+    def test_error_analytics_successful(self, mock_post):
+        send_error_analytics(True)
+        mock_post.assert_called()
+        
+        
+    # as above, send_launch_analytics should probably be a method of Interface
+    @mock.patch("requests.post")
+    def test_launch_analytics_doesnt_crash_on_connection_error(self, mock_post):
+        mock_post.side_effect = requests.ConnectionError()
+        send_launch_analytics(analytics_enabled=True,
+                              inbrowser=True, is_colab="is_colab",
+	                              share="share", share_url="share_url")
+        mock_post.assert_called()
+        
+    @mock.patch("requests.post")
+    def test_launch_analytics_doesnt_post_if_not_enabled(self, mock_post):
+        send_launch_analytics(analytics_enabled=False,
+                              inbrowser=True, is_colab="is_colab",
+	                              share="share", share_url="share_url")
+        mock_post.assert_not_called()
+     
+    @mock.patch("requests.post")   
+    def test_launch_analytics_successful(self, mock_post):
+        send_launch_analytics(analytics_enabled=True,
+                              inbrowser=True, is_colab="is_colab",
+	                              share="share", share_url="share_url")
+        assert mock_post.call_args.kwargs["data"] == {
+	            'launch_method': 'browser',
+	            'is_google_colab': "is_colab",
+	            'is_sharing_on': "share",
+	            'share_url': "share_url",
+	            'ip_address': ip_address
+	        }
+
+    def test_reset_all(self):
+        interface = Interface(lambda input: None, "textbox", "label")
+        interface.close = mock.MagicMock()
+        reset_all()
+        interface.close.assert_called_with()
+        
+    def test_examples_invalid_input(self):
+        with self.assertRaises(ValueError):
+            Interface(lambda input: None, examples=1234)
+        
+    def test_examples_not_valid_path(self):
+        with self.assertRaises(FileNotFoundError):
+            interface = Interface(lambda input: None, "textbox", "label", examples='wrong-path')
+            interface.launch()
+
+    def test_interpretation_shap_not_installed(self):
```

Comment:
If I understand correctly, this would throw an error on any machine where `shap` is installed. In fact, it will throw an error on CircleCI since we install `shap` there for tests. I would remove this or set it up with `mock` such that importing shap throws an ImportError.